gp-settings.php

```diff
@@ -137,11 +142,40 @@ function gp_activate_plugin() {
 }
 register_activation_hook( GP_PLUGIN_FILE, 'gp_activate_plugin' );
 
+/**
+ * Run the plugin de-activation code.
+ *
+ * @since 0.1
+ */
+function gp_deactivate_plugin() {
+	// flush the rewrite rule option so it will be re-generated next time the plugin is activated.
+	update_option( 'gp_rewrite_rule', '' );
```

Comment:
Unfortunately the deactivation hook is called after the re-write rules have already been registered so calling ```flush_rewrite_rules();``` at that point does nothing and there's no way to "delete" the rewrite rules once they've been added (well, no supported way).